docs/layer_util.js

```diff
@@ -610,24 +621,47 @@ function removeScoreLayer() {
   redrawLayers();
 }
 
-const mapLoadingFinished = () => loadingElementFinished(mapContainerId);
+/** Notes that a layer has finished rendering on the map. */
+function mapLoadingFinished() {
+  loadingElementFinished(mapContainerId);
+}
 
 /**
  * Notes that an element has started loading, and add a handler to the Promise
- * to note when it finishes.
+ * to note when it finishes. Also handle errors if the Promise fails.
  * @param {Promise} promise
+ * @param {Object} layerInfoForErrors Data for layer in same format as coming
+ *     from Firestore. Must have `display-name` and `index` properties. Only
+ *     used in case of errors.
  * @return {Promise} wrappedPromise
  */
-function wrapPromiseLoadingAware(promise) {
+function wrapPromiseLoadingAware(promise, layerInfoForErrors) {
   addLoadingElement(mapContainerId);
-  return promise.then(mapLoadingFinished);
+  const {index, 'display-name': displayName} = layerInfoForErrors;
+  return promise
+      .catch((err) => {
+        showError(
+            'Error with layer ' + displayName + ', ' + err,
+            'Error loading layer ' + displayName);
+        $('#' + getCheckBoxRowId(index))
+            .prop(
+                'title',
+                'Error showing layer. If you believe the layer is there, try ' +
```

Comment:
can we persist some error message text in this hover text?

Reply:
I think we've had a general tendency of pretty opaque errors in the user-visible part. I'm open to exposing more details, but would probably put it in the snackbar too if we're doing that. The error if the asset is not found looks like: `Collection asset 'users/gd/1999-janak/data-just-national-block-groups-19' not found.` If you think that would be useful, I can add it.

Reply:
I think it'd be useful, maybe rephrased a tiny bit friendlier like "Underlying earth engine asset '...' not found." Just nice to have something that doesn't disappear (the snackbar fades eventually right?) / you can have multiple of at once.

Reply:
What do you want to happen if the message doesn't end with 'not found.'? Suppress it, or show it verbatim?

Reply:
I'd say show it verbatim since it's just hover text hopefully it won't be too annoying for anyone who can't understand it.

Reply:
Done. Pushed logic down into library function so it could be unified.